### DISCUSSION TITLE
Some Code and Ideas

### DISCUSSION THREAD

**Discussion by emperor06** [dakanji/RefindPlus#188] [General]
Hi Dakanji,

First of all, thank you so much for RefindPlus. It works like a charm, is very customizable, and you made it so easy to compile it's a dream.

I've been using it on several machines for at least a year and made several modifications to suit my needs. Most of them are just personal choices or things that only apply to my use-case but I decided to publish them anyway, in case you find some ideas interesting and suitable for general use.

Let me know if you need any of them. Each modification has its own branch with a verbose commit that explains what it does.

Those are the branches named `feat_*` and `mod_*` on my [repo](https://github.com/emperor06/RefindPlus)

**Comment by dakanji** [comment]
Hi,

Took a quick look and they look interesting.

I do have things a bit up in the air in terms of the code base and it will take quite a bit of time before stuff can be pulled in.
Will get back to you when I have things lined up; which will probably be after the next release.
Apologies, but we are looking at months in terms of time frames.

In the meantime, I have also been working on only having one item shown for each tool.
You can get a preview here: https://temp-file-share.web.app/d/ZV3W-ZYKd
Let me know what you think.

**Comment by dakanji** [comment]
The items suggested have largely been incorporated, often in a different way but materially the same, as of the v14.0.2.AB release.
- feat_confirm
  - Reboot, Shutdown etc no longer require confirmation 
- feat_splash
  - Proposed implementation deemed to be a bit too complicated
  - A similar feature can be implemented using the `BootLogo` feature
    - See notes in config file under `disable_bootlogo`
- feat_ventoytool
  - `handle_ventoy` was already present as noted
  - Additionally, Ventoy is now automatically detected and handled (config token removed)
- mod_flickering
  - Not done as cannot replicate the flickering
    - Already addressed a long time ago although there is still a screen refresh
  - Importantly, disabling all refresh is not desirable as needed in some circumstances to clear the screen
    - That is, it may not be applicable to your setup but is needed on some others.
  - Either way, the "issue" should now be even more inperceptible  
- mod_gfxtools
  - Ugliness is subjective and not sure about being problematic
  - Also, some tools require text mode and a blanket `forced-to-graphics-mode` is undesirable
  - Therefore, a different equivalent has done
    - Users can now add `tools` to `use-graphics-for` for those or `everything` as that says
- mod_onetool
  - Most tools are consolidated into one item displayed in a different way
  - All available tools are accessible but only one icon is shown on the main screen
- mod_quiettools
  - Not done as set out because the output, showing passed parameters, is desired
  - Note however, that anything run in graphcs screen mode, including tools - which can be set via `use_graphics_for`, are "quiet".

**Comment by emperor06** [comment]
Very nice, thanks a lot for the hard-work and feedback. I'll install the new version asap.

**Comment by dakanji** [comment]
I'll take another look at `feat_splash` later.
Seems too linked to `mod_flicker` though.
